src/main/java/forestry/apiculture/ApiaristAI.java

```diff
@@ -0,0 +1,152 @@
+import forestry.core.utils.InventoryUtil;
+import net.minecraft.block.Block;
+import net.minecraft.entity.ai.EntityAIMoveToBlock;
+import net.minecraft.entity.passive.EntityVillager;
+import net.minecraft.inventory.InventoryBasic;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+@SuppressWarnings("static-access")
+public class ApiaristAI extends EntityAIMoveToBlock{
+	private final EntityVillager villager;
+	private boolean hasDrone;
+	private boolean hasPrincess;
+	private InventoryBasic villagerInventory;
+	
+	public ApiaristAI(EntityVillager villager, double speed) {
+		super(villager, speed, 16);
+		this.villager = villager;
+		villagerInventory = villager.getVillagerInventory();
+	}
+	
+	@Override
+	public boolean shouldExecute() {
+		if(this.runDelay<0) {
+			this.hasDrone=hasBeeType(EnumBeeType.DRONE);
+			this.hasPrincess=hasBeeType(EnumBeeType.PRINCESS);
+		}
+		return super.shouldExecute();
+	}
+	
+	@Override 
+	public void updateTask()
+	 {
+        super.updateTask();
+        this.villager.getLookHelper().setLookPosition((double)this.destinationBlock.getX() + 0.5D, (double)(this.destinationBlock.getY() + 1), (double)this.destinationBlock.getZ() + 0.5D, 10.0F, (float)this.villager.getVerticalFaceSpeed());
+
+        if (this.villager.getDistanceSqToCenter(this.destinationBlock) > 4.0D)
+        {
+            World world = this.villager.world;
+            BlockPos blockpos = this.destinationBlock;
+            if(!(TileUtil.getTile(world, blockpos) instanceof TileBeeHouse)) {
+            	return;
+            }
+        	TileBeeHouse beeHouse = (TileBeeHouse) TileUtil.getTile(world,blockpos);
+			InventoryBeeHousing inventory = (InventoryBeeHousing) beeHouse.getBeeInventory();
+            
+			//fill slots from inside bee house
+        	for(ItemStack stack : InventoryUtil.getStacks(inventory, inventory.SLOT_PRODUCT_1, inventory.SLOT_PRODUCT_COUNT)) {
+        		if(stack!=ItemStack.EMPTY && stack.getItem() instanceof ItemBeeGE) {
+        			EnumBeeType type = ((ItemBeeGE)stack.getItem()).getType();
+        			if(inventory.getStackInSlot(inventory.SLOT_QUEEN).isEmpty() && type==EnumBeeType.PRINCESS) {
+        				inventory.setQueen(stack.copy());
+						stack.setCount(0);
+        			} else if(type==EnumBeeType.DRONE) {
+        				InventoryUtil.addStack(inventory, stack, inventory.SLOT_DRONE, 1, true);
+        			}
+        		}
+        	
+        	}
+			
+        	
+        	//fill slots from villager inventory
+        	if(inventory.getStackInSlot(inventory.SLOT_DRONE).isEmpty() || inventory.getStackInSlot(inventory.SLOT_QUEEN).isEmpty()) {
+        		boolean princessAdded = false;
+        		boolean droneAdded = false;
+        		for(ItemStack stack : InventoryUtil.getStacks(villagerInventory)) {
+        			if(princessAdded && droneAdded) {
+        				break;
+        			}
+        			if(stack!=ItemStack.EMPTY && stack.getItem() instanceof ItemBeeGE) {
```

Comment:
Never compare `ItemStack.EMPTY` with `==` or `!=`. You must use `!stack.isEmpty()` here.